akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -36,6 +37,469 @@ import scala.annotation.switch
+          if (pp.keepSeeking) {
+            proceed(pp.nextInput, pp)
+          }
+
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InObject, this)
+
+        case SquareBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InArray, this)
+
+        case _ ⇒
+          pp.take()
+          pp.enterState(InNaked(this, pp))
+
+        //case _ =>
+        //  throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}]")
+      }
+    }
+
+    private object InArray extends InContainerBase(SquareBraceEnd) with LeaveContainerOnExit {
+      override def toString: String = "InArray"
+    }
+    private object InOuterArray extends InContainerBase(SquareBraceEnd) with CompleteObjectAndLeaveContainerOnExit {
+      override def toString: String = "InOuterArray"
+    }
+
+    private object InObject extends InContainerBase(CurlyBraceEnd) with LeaveContainerOnExit {
+      override def toString: String = "InObject"
+    }
+    private object InOuterObject extends InContainerBase(CurlyBraceEnd) with CompleteObjectAndLeaveContainerOnExit {
+      override def toString: String = "InOuterObject"
+    }
+
+    private object NotArrayAfterElement extends ParserState {
+      override def toString: String = "NotArrayAfter"
+
+      /* in this state we know we are not in a JSON array-formatted stream, but we don't yet know yet what kind of
+      separator is being used. We'll never revisit this state or InitialState once we meet a separator
+       */
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = input match {
+        case Comma ⇒
+          pp.skip()
+          pp.enterState(CommaSeparatedBeforeElement)
+
+        case LineBreak ⇒
+          pp.skip()
+          pp.enterState(LinebreakSeparatedBeforeElement)
+
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+
+        /* This is a tolerance; since there is no ambiguity, we can consider a next object immediately after
+          the previous, even without a Comma or Linebreak separator. Akka Stream 2.5.16 has historically supported that.
+
+          We can't extend the same tolerance to arrays, as any stream whose first non-whitespace character is a SquareBraceStart
+          will be considered to be a JSON Array stream
+           */
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterObject, NotArrayAfterElement)
+
+        case _ ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}] — junk after value in linebreak or comma-separated stream")
+      }
+    }
+
+    private abstract sealed class SeparatorSeparatedAfterElement extends ParserState {
+      def separator: Int
+      def separatorName: String
+
+      def beforeNextItem: ParserState
+
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = input match {
+        case _ if input == separator ⇒
```

Comment:
```case `separator` => ```

Reply:
actually… `stable identifier required, but SeparatorSeparatedAfterElement.this.separator found.`

Reply:
(found a way around this)